JT/12_data_structures/data_structure.adoc

```diff
@@ -0,0 +1,76 @@
+= Data Structure
+:toc:
+:toc-title:
+:icons: font
+
+== Introduction to Data Structures
+
+*_What is Data Structure?_* +
+A data structure is a way to store data.
+
+We structure data in different ways depending on what data we have, and what we want to do with it.
+It defines how information can be stored, accessed, and manipulated within a programming language.
+Understanding correlations among data elements allows developers to process information more efficiently.
+
+TIP: The data structures and data types are often used interchangeably.
+But data structures are used to store the various data types.
+
+*_What is needs of Data Structure?_* + :::
+The structure of the data and the synthesis of the algorithm are relative to each other.
+Data presentation must be easy to understand so the developer, as well as the user, can make an efficient implementation of the operation.
+. Data structure modification is easy
+. It requires less time
+. Save storage memory space
+. Data representation is easy
+. Easy access to the large database
+
+image::resources/DS_memes.png[align="center"]
+
+== Advantages and disadvantages of Data Structure
+
+|===
+|Advantages |Disadvantages
+
+|*Efficiency:* If the choice of a data structure for implementing is proper, it makes the program very efficient in terms of time and space.
+|*Implementation complexity:* Some data structures require complex algorithms and implementation methods, which can be difficult to understand and maintain.
+
+|*Reusability:* The data structure provides reusability means that multiple client programs can use the data structure.
+|*Limitations:* Some data structures have limitations on operations or require special conditions for efficient operation, which can complicate usage in certain scenarios.
+
+|*Abstraction:* The data structure also provides the level of abstraction. The client cannot see the internal working of the data structure, so it does not have to worry about the implementation part. The client can only see the interface.
+|*Need for selection:* Choosing the appropriate data structure depends on the specific task, and selecting the wrong one can lead to inefficient program operation.
+|===
+
+== Types of Data Structures
+
+image::resources/ds_types.png[align="center"]
```

Comment:
Probably it would be better to add picture after text. that explaining what is displaying. It is hard to understand picture before reading text. My advice is to add this picture after explaining that In Java, data types are divided into two large groups.

Reply:
I think that diagram needs to be in the top of section because it represent all the types of data and people will understand, what will be written below